pkg/storage/s3/storage.go

```diff
@@ -83,8 +84,21 @@ func (s *StorageProvider) GetVolumeSize(ctx context.Context, volume models.Input
 	if err != nil {
 		return 0, err
 	}
-	size := uint64(0)
+	var size uint64
 	for _, object := range objects {
+		// Check for negative size
+		if object.size < 0 {
+			return 0, fmt.Errorf("invalid negative size for object: %d", object.size)
+		}
+
+		// Check for overflow
+		// MaxUint64 - size = remaining space before overflow
+		//nolint:gosec // G115: negative values already checked
+		if object.size > 0 && uint64(object.size) > math.MaxUint64-size {
+			return 0, fmt.Errorf("total size exceeds uint64 maximum")
+		}
+
+		//nolint:gosec // G115: Already checked above
```

Comment:
nit: it should be safe to skip these as we are basically testing S3 SDK in this case

Reply:
I would like to explicitly leave this if possible due to how we are dealing with a lot of `unint` conversation which may not be needed. 
I added this issue to not forget about it https://github.com/bacalhau-project/bacalhau/issues/4652